### ISSUE TITLE
About teacher-student framework in paper.

### ISSUE THREAD

**Issue by ing907** [willpower057/RankMix#1]
Hello. Thank you for the excellent work and repository; they have been very helpful.

I'm currently attempting to address my issue using RankMix with my own dataset, but I have a question about it.

I noticed that the teacher-student framework, mentioned in the CVPR 2023 paper, does not seem to be implemented in this repository.

Could you please confirm if this code has been implemented? If not, I'm curious whether there are plans to implement it and upload it later.

Thank you.

**Comment by willpower057** [comment]
Hello:

I don't release teacher-student module in this repo for two reasons:
1.  As mentioned in the appendix of paper, the fine-tuning approach is more stable and better (i.e., ```--fix_score_function 'finetune'``` in this repo).
2. Settings are too many and complicated to make it simple.

I don't have the plan to re-implement it, but I can tell you how to achieve it
- Declare the teacher model and make **mixup_by_rank** function use the particular score function:
    - Set ```--fix_score_function 'fixed'/'distill'```, and load your teacher weight as ```pretrained_milnet```. (**train_mil_rankmix.py**)
    - Freeze the weight of ```pretrained_milnet```.
    - Modify **"train/train_frmil"** function of DSMIL/FRMIL classes in dsmil.py/frmil.py (make **mixup_by_rank** function use the particular score function)
        1. Fix the score function, ```score_function = pretrained_milnet```
        2. Distill/finetune the score function, ```score_function = milnet```
- Declare the student model and distill the knowledge
    - (optional) Initialize the ```milnet``` with teacher weight (more stable in our experiment)
    - (optional) If you want to use additional distillation head (DH) mentioned in the appendix
        - Use ```milnet.add_distilled_bag_head``` or ```milnet.add_distilled_score_head``` to add additional heads in dsmil.py/frmil.py
        - Modify **"forward"** function of DSMIL/FRMIL in dsmil.py/frmil.py (output the distilled score/class of the head.)
    - Distill the knowledge
        - Modify **"train/train_frmil"** function of DSMIL/FRMIL in dsmil.py/frmil.py
        - Depend on the distillation loss function you want to use
